app/src/main/java/com/github/orkest/ViewModel/profile/ProfileViewModel.kt

```diff
@@ -97,5 +97,102 @@ open class ProfileViewModel(private val user: String) : ViewModel() {
         }
     }
 
+    /**
+     * Function only called when visiting the profile of another user
+     * Checks whether the current logged in user follows this account
+     */
+    open fun isUserFollowed(): CompletableFuture<Boolean>{
+        if(user == Constants.currentLoggedUser){ Log.e(TAG, "Cannot call this function when visiting the current logged-in user's profile", IllegalArgumentException()) }
+        val future = CompletableFuture<Boolean>()
+        userDocument(user).get()
+            .addOnSuccessListener { document ->
+                if (document != null && document.exists()) {
+                    val user = document.toObject(User::class.java)
+                    if (user != null && user.followers.contains(Constants.currentLoggedUser)) {
+                        future.complete(true)
+                    } else{
+                        future.complete(false)
+                    }
+                }
+            }
+            .addOnFailureListener { e -> Log.w(TAG, "Error getting user data", e) }
+        return future
+    }
+
+    open fun follow(): CompletableFuture<Boolean>{
+        val futureFollow =CompletableFuture<Boolean>()
+
+        val userUpdated = updateUser(toFollow = true)
+        val currentUserUpdated = updateCurrentUser(toFollow = true)
+
+        //Both updates must be successful
+        futureFollow.complete(userUpdated && currentUserUpdated)
+        return futureFollow
+    }
+
+    open fun unfollow(): CompletableFuture<Boolean>{
+        val futureFollow = CompletableFuture<Boolean>()
+
+        val userUpdated = updateUser(toFollow = false)
+        val currentUserUpdated = updateCurrentUser(toFollow = false)
+
+        //Both updates must be successful
+        futureFollow.complete(userUpdated && currentUserUpdated)
+        return futureFollow
+    }
+
+    /**
+     * Updates the user's followers' list
+     * toFollow: Boolean = represents whether or not the current logged in user wants to follow this account
+     */
+    private fun updateUser(toFollow: Boolean): Boolean{

```

Comment:
This function is similar to the updateCurrentUser, maybe you could refactor the successListener into another private method with a boolean to inform wether it is an update for the followers or followings

Reply:
I am accessing two different paths in the database. I implemented it that way the first time but I thought it wasn't really readible. But I can do that in future commits if you think it is a better option